server/app/filters/DisableCachingFilter.java

```diff
@@ -22,11 +34,28 @@ public EssentialAction apply(EssentialAction next) {
         request ->
             next.apply(request)
                 .map(
-                    result ->
-                        result
-                            .withHeader("Cache-Control", "no-cache, must-revalidate")
-                            .withHeader("Pragma", "no-cache")
-                            .withHeader("Expires", "0"),
+                    result -> {
+                      Optional<String> pathPrefix = Iterables.getFirst(PATH_SPLITTER.splitToList(request.uri()));
+                      Integer status = result.status();
+
+                      if (environment.isDev()) {
+                        // Must revalidate status asset caches in dev mode
+                        // pathPrefix = Optional.empty();
+                      }
+                      if (pathPrefix.map(ASSET_PATH_PREFIXES::contains).orElse(false)) {
+                        // Only cache when Status is OK https://web.dev/uses-long-cache-ttl/
+                        if (status == 200 || status == 203 || status == 206) {
```

Comment:
suggestion (blocking): use a single `if`.

(optional) could use another constant: `OK_STATUS_CODES = ImmutableSet.of(200, 203, 206);` and then

```java
private static final ImmutableSet<Integer> OK_STATUS_CODES = ImmutableSet.of(200, 203, 206);
...
if (pathPrefix.map(ASSET_PATH_PREFIXES::contains).orElse(false)
    // Only cache when Status is OK https://web.dev/uses-long-cache-ttl/
    && OK_STATUS_CODES.contains(status)) {
```